shenyu-web/src/main/java/org/apache/shenyu/web/loader/ShenyuPluginLoader.java

```diff
@@ -106,7 +99,12 @@ public List<ShenyuLoaderResult> loadExtendPlugins(final String path) throws IOEx
             return Collections.emptyList();
         }
         List<ShenyuLoaderResult> results = new ArrayList<>();
+        boolean loadNewPlugin = false;
         for (File each : jarFiles) {
+            if (jars.stream().map(PluginJar::absolutePath).filter(StringUtils::hasText).anyMatch(p -> p.equals(each.getAbsolutePath()))) {
```

Comment:
From my point of view, the `absolutePath` is just able to avoid load the plugin twice.

Reply:
Once the plugin has been loaded, it's not unnecessary to be added in the `jars` list.

Reply:
If you put a new plugin with the same absolute path, it may be not loaded, but this case is seem not recommended.

Reply:
Hi， thanks.


What about version upgrades, how should we handler?

Reply:
IMO, it might be a new feature. 
When version upgrades, we can replace the previous jars with the new one. What's more, we should destroy the previous classes that are already loaded in spring context and then registered the new classes.

Reply:
new version have same absolutePath？

Reply:
Base on `v2.5.0`, the plugin of new version with the same absolutePath will also not be loaded if the className is not changed. 
![image](https://user-images.githubusercontent.com/15926669/207218707-496b31b6-8c1c-4d59-a1de-bfe987f96615.png)
If we want to handle this,  should we make a new feature to solve it?

Reply:
Or if you worry about the `absolutePath` way, maybe using hash of the ext-plugin jar is a better way.

Reply:
yes，we should resolve the version update.

Looking forward to your next pr